### PR TITLE
[PTT-238] add paVerifyPaymentNotice

### PR DESCRIPTION
**Pull request by pasqualespica**
#### List of Changes

- added 
  - `PaaErrorEnum` enum according to [pagoPa fault code](https://docs.google.com/spreadsheets/d/15sJplTLu6kWPomW3jbBzZYvD7QyIswERKktFG0bk1IQ/edit#gid=71375386)
  - `PartnerServiceTest` unit test suite
- updated
  - `.env` payments file with
    - `PAA_ID_DOMINIO`
    - `PAA_ID_INTERMEDIARIO`
    - `PAA_STAZIONE_INT`
  - `PartnerService` with `PaVerifyPaymentNoticeRes` implementation
 
- inserted
  - `PaymentOptionsRepository` to get DB `payments_option` access

#### Motivation and Context

The goal of this PR is to add `paVerifyPaymentNotice`'s behavior in order to allow the PTT portal to put the primitive at the service of the payment node, according to **pagoPA** [paVerifyPaymentNotice](https://pagopa.github.io/pagopa-api/indexPA.html#operation/paVerifyPaymentNotice) specifications
 
#### How Has This Been Tested?

Run tests typing from console : 
```
mvn clean verify
```

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: payments/pom.xml [hunk 1]
```diff
@@ -141,14 +141,14 @@
             </exclude>
           </excludes>
         </configuration>
-        <executions>
+        <!-- <executions>
```
Comment (GiovanniMancini) [inline payments/pom.xml:144]: We'd better remove this old code

File: payments/src/test/java/it/gov/pagopa/hubpa/payments/service/PartnerServiceTest.java [hunk 2]
```diff
@@ -1,70 +1,237 @@
...
+    when(factory.createCtPaymentOptionsDescriptionListPA())
+        .thenReturn(factoryUtil.createCtPaymentOptionsDescriptionListPA());
+
+    when(paymentOptionRepository.findByNotificationCode(requestBody.getQrCode().getNoticeNumber()))
+        .thenReturn(Optional.of(DebitorMock.createPaymentOptionsMock4()));
+
+    // Test execution
+    PaVerifyPaymentNoticeRes responseBody = partnerService.paVerifyPaymentNotice(requestBody);
+
+    // Test postcondiction
```
Comment (GiovanniMancini) [inline payments/src/test/java/it/gov/pagopa/hubpa/payments/service/PartnerServiceTest.java:81]: Just a typo

**Review by GiovanniMancini — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]
Well done!

**Comment by GiovanniMancini** [comment]
@pasqualespica could you please rebase with the main branch?

**Pull request merged**